packages/transport-tcp/src/listener.ts

```diff
@@ -195,11 +196,13 @@ export class TCPListener extends TypedEventEmitter<ListenerEvents> implements Li
     }
 
     this.log('new inbound connection %s', maConn.remoteAddr)
+    this.maConnections.add(maConn)
 
     this.context.upgrader.upgradeInbound(maConn)
       .then((conn) => {
         this.log('inbound connection upgraded %s', maConn.remoteAddr)
         this.connections.add(maConn)
+        this.maConnections.delete(maConn)
```

Comment:
The `maConn` is only deleted from the set after a successful upgrade, if the upgrade fails it stays in the set until the transport is closed - this could be many hours, days or months after the transport was started so it introduces a memory leak.

Reply:
Ah this is a problem; maConn should be removed from maConnections inside of the catch block.